### ISSUE TITLE
rvm fails to install 2.4.1 on Ubuntu 17.10

### ISSUE THREAD

**Issue by retorquere** [rvm/rvm#4236]
## Description

rvm fails to install 2.4.1 on Ubuntu 17.10

## Steps to reproduce

`rvm install ruby-2.4.1` yields

```
Searching for binary rubies, this might take some time.
No binary rubies available for: ubuntu/17.10/x86_64/ruby-2.4.1.
Continuing with compilation. Please read 'rvm help mount' to get more information on binary rubies.
Checking requirements for ubuntu.
Installing requirements for ubuntu.
Updating system....
Installing required packages: libreadline6-dev...
Requirements installation successful.
Installing Ruby from source to: /usr/share/rvm/rubies/ruby-2.4.1, this may take a while depending on your cpu(s)...
ruby-2.4.1 - #downloading ruby-2.4.1, this may take a while depending on your connection...
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
  0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0Warning: Failed to create the file ruby-2.4.1.tar.bz2.part: Permission denied
  0 11.9M    0  2199    0     0   2199      0  1:35:16 --:--:--  1:35:16 16915
curl: (23) Failed writing body (0 != 2199)
There was an error(23).
Checking fallback: https://ftp.ruby-lang.org/pub/ruby/2.4/ruby-2.4.1.tar.bz2
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
  0     0    0     0    0     0      0      0 --:--:--  0:00:01 --:--:--     0Warning: Failed to create the file ruby-2.4.1.tar.bz2.part: Permission denied
  0 11.9M    0 16149    0     0  16149      0  0:12:58  0:00:01  0:12:57  9206
curl: (23) Failed writing body (0 != 16149)
There was an error(23).
Failed download
There has been an error fetching the ruby interpreter. Halting the installation.
```

## Environment info

```

system:

  system:
    uname:        "Linux celery 4.13.0-16-generic #19-Ubuntu SMP Wed Oct 11 18:35:14 UTC 2017 x86_64 x86_64 x86_64 GNU/Linux"
    name:         "Ubuntu"
    version:      "17.10"
    architecture: "x86_64"
    bash:         "/bin/bash => GNU bash, version 4.4.12(1)-release (x86_64-pc-linux-gnu)"
    zsh:          " => not installed"
    remote_path:  "ubuntu/17.10/x86_64"

  rvm:
    version:      "1.29.3 (manual)"
    updated:      "5 minutes 45 seconds ago"
    path:         "/usr/share/rvm"
    autolibs:     "[4] Allow RVM to use package manager if found, install missing dependencies, install package manager (only OS X)."

  homes:
    gem:          "not set"
    ruby:         "not set"

  binaries:
    ruby:         ""
    irb:          ""
    gem:          ""
    rake:         ""

  environment:
    PATH:         "/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin:/usr/games:/usr/local/games:/snap/bin:/usr/share/rvm/bin"
    GEM_HOME:     ""
    GEM_PATH:     ""
    MY_RUBY_HOME: ""
    IRBRC:        ""
    RUBYOPT:      ""
    gemset:       ""
```

**Comment by pkuczynski** [comment]
Seems that you have no write permissions to the location where you try to install:

```
Failed to create the file ruby-2.4.1.tar.bz2.part: Permission denied
```

Could you check that?

**Comment by retorquere** [comment]
How can I tell which location rvm is trying to access? I have write access to the directory I run the command in.

Aaaand of course now I try again, it just works. Same directory. I swear.

**Comment by pkuczynski** [comment]
Hmm, hard to debug then. Shall we close? Re-open if you find something...

**Comment by retorquere** [comment]
Closing is fine - I generally don't want to close issues myself because I want to let the dev make the assessment if everything is done (it's how I prefer issues on my own repos be handled)

**Comment by raelgc** [comment]
@retorquere Do you mind to try the [Ubuntu package](https://github.com/rvm/ubuntu_rvm)? Usually it helps with this kind of issue.

**Comment by retorquere** [comment]
I did install from RPM :)

**Comment by raelgc** [comment]
RPM?

**Comment by retorquere** [comment]
Eh, deb, sorry. Been spending the day with centos unfortunately.

**Comment by trickymuffin** [comment]
Hi @pkuczynski @raelgc  I'm having the same issue, (ubuntu 17.10) I can't figure out where is rvm trying to download the package so I can give the proper permissions to the directory

**Comment by raelgc** [comment]
Had you previously installed `rvm` manually or ran any `rvm` commando using `sudo`? In case of yes, please, try remove any previous install removing the package then running `rvm` implode.

**Comment by SkyWriter** [comment]
RVM 1.29.3 from Ubuntu package here. It tried to download an archive to `/usr/share/rvm/archives`, but failed to do so due to lack of permissions.

The issue ended up being my SSH connection being cached. Even though I seemingly closed my current SSH session and reconnected, the actual session didn't get reinstantinated, hence my user didn't pick up the new "rvm" group and consequently didn't have the right permission.

Hope this helps some future weary traveller.

**Comment by nebulousdog** [comment]
I had this issue too, but I'm not doing anything via ssh afaik like @SkyWriter described when downloading.

```
rvm install ruby-2.4.4
Searching for binary rubies, this might take some time.
No binary rubies available for: ubuntu/19.04/x86_64/ruby-2.4.4.
Continuing with compilation. Please read 'rvm help mount' to get more information on binary rubies.
Checking requirements for ubuntu.
Requirements installation successful.
Installing Ruby from source to: /usr/share/rvm/rubies/ruby-2.4.4, this may take a while depending on your cpu(s)...
ruby-2.4.4 - #downloading ruby-2.4.4, this may take a while depending on your connection...
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
  0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0Warning: Failed to create the file ruby-2.4.4.tar.bz2.part: Permission denied
  0 12.0M    0  2750    0     0  58510      0  0:03:36 --:--:--  0:03:36 58510
curl: (23) Failed writing body (0 != 2750)
There was an error(23).
Checking fallback: https://ftp.ruby-lang.org/pub/ruby/2.4/ruby-2.4.4.tar.bz2
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
  0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0Warning: Failed to create the file ruby-2.4.4.tar.bz2.part: Permission denied
  0 12.0M    0  2750    0     0  67073      0  0:03:08 --:--:--  0:03:08 68750
curl: (23) Failed writing body (0 != 2750)
There was an error(23).
Failed download
There has been an error fetching the ruby interpreter. Halting the installation.
```
